### PR TITLE
Simplify semanticTypeInfo() by replacing Visitor

### PR DESCRIPTION
**Pull request by WalterBright**
Much of the diffs are whitespace indenting changes.

### CODE REVIEW

File: src/dmd/dstruct.d [hunk 1]
```diff
@@ -69,137 +69,117 @@ extern (C++) FuncDeclaration search_toString(StructDeclaration sd)
...
+        else
         {
-            t.basetype.accept(this);
+            getTypeInfoType(sd.loc, t, sc);
+            sd.requestTypeInfo = true;
+
+            // https://issues.dlang.org/show_bug.cgi?id=15149
+            // if the typeid operand type comes from a
+            // result of auto function, it may be yet speculative.
+            unSpeculative(sc, sd);
```
Comment (rainers) [inline src/dmd/dstruct.d]: This line was commented before. Is it changed deliberately? During a refactoring?

Comment (WalterBright) [reply src/dmd/dstruct.d]: Dang, I just wasted the last hour finding that one! (I don't know how it got uncommented.)

File: src/dmd/dstruct.d [hunk 2]
```diff
@@ -69,137 +69,117 @@ extern (C++) FuncDeclaration search_toString(StructDeclaration sd)
...
-        {
-        }
-
-        override void visit(TypeTuple t)
+    void visitTuple(TypeTuple t)
+    {
+        if (t.arguments)
         {
-            if (t.arguments)
+            foreach (arg; *t.arguments)
```
[the remark this answers is not in the document]
Comment (WalterBright) [reply src/dmd/dstruct.d]: It actually is done the way you suggest.

File: src/dmd/dstruct.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/dstruct.d]: (I probably fat-fingered my editor.)

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**